2023/incoming/html-xss-scrubbing-and-file-upload-validation-in-catalyst.pod

```diff
@@ -0,0 +1,67 @@
+Author: David Precious <davidp@preshweb.co.uk>
+Title: HTML/XSS scrubbing and file upload validation in Catalyst
+Topic: Catalyst::Plugin::CheckFileUploadTypes
+
+=encoding utf8
+
+At work, we needed to tighten up the security of our Catalyst-powered
+API, with two main requirements:
+
+=for list
+* Stripping HTML/XSS attempts from incoming parameters
+* Validating file uploads are expected and are the expected type
+
+=cut
+
+=head2 Catalyst::Plugin::HTML::Scrubber
+
+We found L<Catalyst::Plugin::HTML::Scrubber>, which at first glance looked
+like it would do at least most of what we needed, automatically scrubbing
+parameters using L<HTML::Scrubber>.
+
+We're not a fan of reinventing wheels when we can avoid it, so I set about
+adding the extra features we needed - in particular, being able to exempt
+particular parameters from scrubbing, by name or regex match - and raised a pull
+request to share that upstream.  Unfortunately, the original author doesn't seem
+to be active in the Perl community any more, and several attempts of contact
+failed - so I approached our friendly CPAN admins for help, obtained co-maint,
+and released a new version.  Since then we added more - including recursive 
```

Comment:
Might be helpful to link to https://neilb.org/2013/07/24/adopt-a-module.html here.